kfs-ld/src/main/java/edu/arizona/kfs/module/ld/batch/service/impl/LaborEncumbranceAdjustmentServiceImpl.java

```diff
@@ -0,0 +1,626 @@
+package edu.arizona.kfs.module.ld.batch.service.impl;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.sql.Date;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.coa.businessobject.ObjectCode;
+import org.kuali.kfs.gl.GeneralLedgerConstants;
+import org.kuali.kfs.gl.batch.service.AccountingCycleCachingService;
+import org.kuali.kfs.module.ld.businessobject.LaborOriginEntry;
+import org.kuali.kfs.module.ld.businessobject.LaborOriginEntryFieldUtil;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSPropertyConstants;
+import org.kuali.kfs.sys.Message;
+import org.kuali.kfs.sys.batch.service.WrappingBatchService;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
+import org.kuali.rice.kim.api.identity.Person;
+import org.kuali.rice.kim.api.identity.PersonService;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.arizona.kfs.module.ld.batch.dataaccess.LaborEncumbranceAdjustmentDao;
+import edu.arizona.kfs.module.ld.batch.dataaccess.LaborEncumbranceDao;
+import edu.arizona.kfs.module.ld.batch.report.LaborEncumbranceJobReportData;
+import edu.arizona.kfs.module.ld.batch.report.LaborEncumbranceReportWriterService;
+import edu.arizona.kfs.module.ld.batch.service.LaborEncumbranceAdjustmentService;
+
+/**
+ * KITT-933 / FP-INT0008-01 - Main service which calculates the needed adjustments.  Used by the related batch steps. 
+ * UAF-4010 MOD-FP0008-01 Accounting for Personnel Encumbrances - Code Feature
+ * 
+ * @author Jonathan Keller
+ */
+@Transactional
+public class LaborEncumbranceAdjustmentServiceImpl implements LaborEncumbranceAdjustmentService {
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(LaborEncumbranceAdjustmentServiceImpl.class);
+
+    protected LaborEncumbranceAdjustmentDao laborEncumbranceAdjustmentDao;
+    protected LaborEncumbranceDao laborEncumbranceDao;
+    protected DateTimeService dateTimeService;
+    protected LaborEncumbranceReportWriterService reportWriterService;
+    protected PersonService personService;
+    protected AccountingCycleCachingService accountingCycleCachingService;
+    protected ParameterService paramSrv;
+        
+
+    /**
+     * @see edu.arizona.kfs.module.ld.batch.service.LaborEncumbranceAdjustmentService#buildBalanceFile(java.lang.Integer, java.io.File)
+     */
+    public boolean buildBalanceFile(Integer fiscalYear, File outputFile) {
+        int outputRecords = laborEncumbranceAdjustmentDao.buildFileForEncumbranceBalances(fiscalYear, outputFile);
```

Comment:
Does the RHS mutate anything of consequence that's needed? If so, I'd suggest to leave just the RHS and remove the LHS; otherwise it introduces doubt due to the dead LHS.